api/apiclient.go

```diff
@@ -166,6 +180,21 @@ func Open(info *Info, opts DialOpts) (*State, error) {
 	return st, nil
 }
 
+// OpenWithVersion uses an explicit version of the Admin facade to call Login
+// on.  This allows the caller to pretends to be an older client, and is used
+// only in testing.
+func OpenWithVersion(info *Info, opts DialOpts, loginVersion int) (*State, error) {
+	switch loginVersion {
+	case 0:
+		return open(info, opts, (*State).loginV0)
```

Comment:
Nit: given that the case blocks are all the same except for the login method arg, I'd prefer to see the login method assigned to a variable in each case block, with `open` called just once at the bottom of the func.

Not a biggie though.